src/plan-integration.ts

```diff
@@ -9,16 +9,92 @@ import path from 'path'
 import os from 'os'
 import * as exec from '@actions/exec'
 
+import * as github from '@actions/github'
+
+function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function waitBuild(githubToken: string): Promise<void> {
+  const octokit = github.getOctokit(githubToken)
+
+  while (true) {
+    await sleep(5000)
+    const jobs = await octokit.paginate(
+      octokit.rest.actions.listJobsForWorkflowRunAttempt,
+      {
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        run_id: github.context.runId,
+        attempt_number: github.context.runAttempt,
+        per_page: 100
+      }
+    )
+    const targetJobs = jobs.filter(j => (j.name || '').includes('Build'))
+    if (targetJobs.length === 0) {
+      core.info('no build jobs')
+      return
+    }
+    let successes = 0
+    core.info('waiting for build jobs:')
+    for (const job of targetJobs) {
+      if (job.status == 'completed') {
+        if (job.conclusion == 'success') {
+          core.info(`[SUCCESS] ${job.name}`)
+          successes++
+        } else if (job.conclusion == 'skipped') {
+          core.info(`[SKIPPED] ${job.name}`)
+          successes++
+        } else {
+          throw new Error(
+            `build job ${job.name} failed with conclusion: ${job.conclusion}`
+          )
+        }
+      } else {
+        core.info(`[${job.status.toUpperCase()}] ${job.name}`)
+      }
+    }
+    if (targetJobs.length === successes) {
+      return
+    }
+  }
+}
+
+async function downloadArtifact(
```

Comment:
I thought this logic was already implemented somewhere else in the pipeline

Reply:
Unfortunately, no. Usually when an artifact download fails, the workflow simply errors out because downloads rarely fail. But in this case it needs to be retried due to a race condition caused by not using the official job-dependencies system.